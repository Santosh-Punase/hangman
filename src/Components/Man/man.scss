@import 'src/assets/scss/variables.scss';

.hangman__man {
  position: relative;
  margin-top: 2rem;
  margin-left: 2rem;
  width: 20rem;
  display: flex;
  flex-direction: column;
  align-items: center;

  &-neck {
    width: 1rem;
    height: 2rem;
    border-bottom-left-radius: 0.3rem;
    border-bottom-right-radius: 0.3rem;
    background-color: $yellow-light;
    margin-top: -1rem;
  }

  &-corpus {
    position: relative;
    width: 4rem;
    height: 5rem;
    margin-top: -0.5rem;
    border-radius: 0.2rem;
    background-color: $green-light;
    &::before {
      content: '';
      width: 2rem;
      height: 5rem;
      position: absolute;
      right: 0;
      top: 0;
      border-top-right-radius: 0.2rem;
      background-color: $green-dark;
    }
    &::after {
      content: '';
      width: 4rem;
      height: 2rem;
      position: absolute;
      right: 0;
      bottom: 0;
      border-bottom-left-radius: 0.2rem;
      border-bottom-right-radius: 0.2rem;
      background-color: $blue-sky-blue;
    }
  }

  &-arm__left {
    position: absolute;
    width: 1.5rem;
    height: 4rem;
    border-bottom-left-radius: 0.2rem;
    border-bottom-right-radius: 0.2rem;
    background-color: $green-light;
    transform: rotate(45deg);
    transform-origin: 0% 0%;
    z-index: 2;
  }

  &-arm__right {
    position: absolute;
    width: 1.5rem;
    left: 2.5rem;
    top: 0;
    height: 4rem;
    border-bottom-left-radius: 0.2rem;
    border-bottom-right-radius: 0.2rem;
    background-color: $green-dark;
    transform: rotate(-45deg);
    transform-origin: 100% 0%;
    z-index: 2;
  }

  &-hand__left {
    position: absolute;
    width: 1rem;
    height: 4.5rem;
    margin: 0.12rem;
    border-bottom-left-radius: 0.2rem;
    border-bottom-right-radius: 0.2rem;
    background-color: $yellow-light;
    transform: rotate(45deg);
    transform-origin: 0% 0%;
    z-index: 1;
  }
  &-hand__right {
    position: absolute;
    right: 0;
    top: 0.12rem;
    width: 1rem;
    height: 4.37rem;
    margin: 0.12rem;
    border-bottom-left-radius: 0.2rem;
    border-bottom-right-radius: 0.2rem;
    background-color: $yellow-light;
    transform: rotate(-45deg);
    transform-origin: 100% 0%;
    z-index: 1;
  }

  &-leg__left {
    position: absolute;
    width: 1.75rem;
    height: 5rem;
    top: 4.5rem;
    left: 0;
    border-bottom-left-radius: 0.2rem;
    border-bottom-right-radius: 0.2rem;
    background-color: $blue-sky-blue;
    transform: rotate(10deg);
    transform-origin: 0% 0%;
    z-index: 2;
  }

  &-leg__right {
    position: absolute;
    width: 1.75rem;
    right: 0;
    top: 4.5rem;
    height: 5rem;
    border-bottom-left-radius: 0.2rem;
    border-bottom-right-radius: 0.2rem;
    background-color: $blue-sky-blue;
    transform: rotate(-10deg);
    transform-origin: 100% 0%;
    z-index: 2;
  }

  &-foot__left {
    position: absolute;
    width: 3rem;
    height: 1rem;
    top: 9rem;
    left: -2rem;
    border-radius: 0.4rem;
    background-color: $yellow-light;
    transform: rotate(10deg);
    transform-origin: 0% 0%;
    z-index: 1;
  }
  &-foot__right {
    position: absolute;
    right: -2rem;
    width: 3rem;
    height: 1rem;
    top: 9rem;
    border-radius: 0.4rem;
    background-color: $yellow-light;
    transform: rotate(-10deg);
    transform-origin: 100% 0%;
    z-index: 1;
  }
}