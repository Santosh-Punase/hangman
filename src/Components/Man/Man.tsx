import React from 'react';

import { Bar } from '../Bar/Bar';
import { Head } from './Components';

import './man.scss';

interface Props {
  lives: number;
}

const Man: React.FC<Props> = ({ lives }) => {
  
  return (
    <div className="hangman__man">
      <Bar />
      {lives <= 10 && <Head />}
      {lives <= 9 && <div className="hangman__man-neck"></div>}
      {lives <= 8 &&
        <div className="hangman__man-corpus">
          {lives <= 7 && <div className="hangman__man-arm__right" />  }
          {lives <= 6 && <div className="hangman__man-arm__left" /> }
          {lives <= 5 && <div className="hangman__man-hand__right" /> }
          {lives <= 4 && <div className="hangman__man-hand__left" />}
          {lives <= 3 && <div className="hangman__man-leg__right" />  }
          {lives <= 2 && <div className="hangman__man-leg__left" /> }
          {lives <= 1 && <div className="hangman__man-foot__right" /> }
          {lives <= 0 && <div className="hangman__man-foot__left" />}
        </div>
      }      
    </div>

  );
}

export { Man };
