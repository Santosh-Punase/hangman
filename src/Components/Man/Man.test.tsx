import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Man } from './Man';

afterEach(cleanup);

describe('Hangman Man should render with', () => {
  [11,10,9,8,7,6,5,4,3,2,1,0].forEach(live => {
    test(`${live} lives`, () => {
      const { asFragment } = render(<Man lives={live} />);
      
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
