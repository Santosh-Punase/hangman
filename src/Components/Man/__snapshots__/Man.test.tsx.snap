// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hangman Man should render with 0 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    >
      <div
        class="hangman__man-arm__right"
      />
      <div
        class="hangman__man-arm__left"
      />
      <div
        class="hangman__man-hand__right"
      />
      <div
        class="hangman__man-hand__left"
      />
      <div
        class="hangman__man-leg__right"
      />
      <div
        class="hangman__man-leg__left"
      />
      <div
        class="hangman__man-foot__right"
      />
      <div
        class="hangman__man-foot__left"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 1 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    >
      <div
        class="hangman__man-arm__right"
      />
      <div
        class="hangman__man-arm__left"
      />
      <div
        class="hangman__man-hand__right"
      />
      <div
        class="hangman__man-hand__left"
      />
      <div
        class="hangman__man-leg__right"
      />
      <div
        class="hangman__man-leg__left"
      />
      <div
        class="hangman__man-foot__right"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 2 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    >
      <div
        class="hangman__man-arm__right"
      />
      <div
        class="hangman__man-arm__left"
      />
      <div
        class="hangman__man-hand__right"
      />
      <div
        class="hangman__man-hand__left"
      />
      <div
        class="hangman__man-leg__right"
      />
      <div
        class="hangman__man-leg__left"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 3 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    >
      <div
        class="hangman__man-arm__right"
      />
      <div
        class="hangman__man-arm__left"
      />
      <div
        class="hangman__man-hand__right"
      />
      <div
        class="hangman__man-hand__left"
      />
      <div
        class="hangman__man-leg__right"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 4 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    >
      <div
        class="hangman__man-arm__right"
      />
      <div
        class="hangman__man-arm__left"
      />
      <div
        class="hangman__man-hand__right"
      />
      <div
        class="hangman__man-hand__left"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 5 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    >
      <div
        class="hangman__man-arm__right"
      />
      <div
        class="hangman__man-arm__left"
      />
      <div
        class="hangman__man-hand__right"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 6 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    >
      <div
        class="hangman__man-arm__right"
      />
      <div
        class="hangman__man-arm__left"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 7 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    >
      <div
        class="hangman__man-arm__right"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 8 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
    <div
      class="hangman__man-corpus"
    />
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 9 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
    <div
      class="hangman__man-neck"
    />
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 10 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
    <div
      class="hangman__man-head"
    >
      <div
        class="hangman__man-ears"
      />
      <div
        class="hangman__man-face"
      />
      <div
        class="hangman__man-forehead"
      />
      <div
        class="hangman__man-eyes eye__left"
      />
      <div
        class="hangman__man-eyes eye__right"
      />
      <div
        class="hangman__man-chick chick__left"
      />
      <div
        class="hangman__man-chick chick__right"
      />
      <div
        class="hangman__man-nose"
      />
      <div
        class="hangman__man-mouth"
      />
      <div
        class="hangman__man-earing"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Hangman Man should render with 11 lives 1`] = `
<DocumentFragment>
  <div
    class="hangman__man"
  >
    <div
      class="hangman__bar-container"
    >
      <div
        class="hangman__bar bar-horizontal"
      />
      <div
        class="hangman__bar bar-vertical"
      />
    </div>
  </div>
</DocumentFragment>
`;
